### PR TITLE
Redirect to the registration page when landing with ?email=

### PR DESCRIPTION
**Pull request by Kegsay**
Fixes "Missing access token" error. https://github.com/vector-im/vector-web/issues/797

### CODE REVIEW

File: src/components/structures/MatrixChat.js [hunk 1]
```diff
@@ -655,7 +667,7 @@ module.exports = React.createClass({
             }
         }
         else {
-            if (screen) console.error("Unknown screen : %s", screen);
+            console.error("Unknown screen : %s", screen);
```
Comment (dbkr) [inline src/components/structures/MatrixChat.js]: Was this intentional?

File: src/components/structures/MatrixChat.js [hunk 2]
```diff
@@ -802,7 +814,6 @@ module.exports = React.createClass({
         var RoomDirectory = sdk.getComponent('structures.RoomDirectory');
         var MatrixToolbar = sdk.getComponent('globals.MatrixToolbar');
         var ForgotPassword = sdk.getComponent('structures.login.ForgotPassword');
-
```
Comment (dbkr) [inline src/components/structures/MatrixChat.js]: Also this

**Comment by dbkr** [comment]
other than some things that looks accidentally committed, lgtm

File: src/components/structures/MatrixChat.js (hunk 1 above)
Comment (Kegsay) [inline src/components/structures/MatrixChat.js]: Yes. I don't feel that we should be suppressing this message if `screen` is `undefined`. We should *never* be calling it with `undefined`.

File: src/components/structures/MatrixChat.js (hunk 2 above)
Comment (Kegsay) [inline src/components/structures/MatrixChat.js]: done

**Comment by Kegsay** [comment]
@dbkr PTAL

File: src/components/structures/MatrixChat.js (hunk 1 above)
Comment (dbkr) [inline src/components/structures/MatrixChat.js]: Seems reasonable.

**Comment by dbkr** [comment]
lgtm

**Pull request merged**